### ISSUE TITLE
Support for Ctrl+Z

### ISSUE THREAD

**Issue by joeyaiello** [PowerShell/PowerShell#3229]
This is strongly tied to #716, but we want to make sure these have independent work items. 

Steps to reproduce
------------------
1. Open PowerShell on Linux/macOS
2. Hit Ctrl+Z

Expected behavior
-----------------
Current running process is forked to bg

Actual behavior
---------------
PowerShell itself is forked to bg and the user is bounced back out to bash (or whatever is the default shell that handled Ctrl+Z)

**Comment by lypanov** [comment]
❤️  One of the only issues that I'm having with using pwsh on Linux currently.

**Comment by arvidcarlandercompose** [comment]
I look forward to a fix - this is a highly useful linux feature.

Is there a workaround or setting in powershell that would make it catch the ctrl-z rather than pass it back to bash?

**Comment by vexx32** [comment]
@arvidcarlandercompose you might try adding a PSRL key handler: `Set-PSReadLineKeyHandler -Chord Ctrl+Z -ScriptBlock { <# some action #> }`

**Comment by superbonaci** [comment]
I guess you can use this script in the meantime:
[Pause-Process for PowerShell](https://github.com/besimorhino/Pause-Process)

By the way, does control+z actually pause the process (IO included), or only sends it to background?

**Comment by superbonaci** [comment]
You can use also [PsSuspend](https://docs.microsoft.com/en-us/sysinternals/downloads/pssuspend) and put it into your PATH, or use [Process Explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer), which as a GUI to suspend/resume processes.

**Comment by JordanShurmer** [comment]
This feature would be really great to have.

In bash I frequently use this while developing.. I'd start a server and watch the logs at first, then when I want to do other things I'd simply hit Ctrl-Z and it would keep running in the background.

In Bash: `^z` to suspend the current job. `bg` to resume it.

Here's the docs about [Job Control in GNU Bash](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html) for all the details.

**Comment by KhabarlakKonstantin** [comment]
Having Ctrl+Z process suspension in Windows would be also very useful

**Comment by deekshithanand** [comment]
its been 3 years keeping the request alive. I recently migrated from linux to windows, and really would love to have this feature! Please!

**Comment by Lite5h4dow** [comment]
still waiting for this ⌚ ...

**Comment by yeahnoob** [comment]
Oh My .. each time after I pressed Ctrl+Z in my muscle memory.

**Comment by 4zv4l** [comment]
Is there any news about that feature ? T-T

**Comment by powercode** [comment]
There is a substantial difference in the underlying model between the *nix systems, which start new processes for each part of the pipeline, and PowerShell, which runs the pipeline on a thread in a shared memory space unless there is a native command involved, in which case you get a mix of behaviors. 

I'm not saying that it is impossible, but the more independent things you have running in your address space, the higher the risk for unintended consequences. 

I think this is a price we pay for all the benefits we get from having rich objects in the pipeline.

So don't hold your breath waiting for this 😉

**Comment by superbonaci** [comment]
Maybe Ctrl+Z is assigned to another function in Windows...

**Comment by superbonaci** [comment]
Is it done or closed due to inactivity?

**Comment by Lavoista** [comment]
any hope???

**Comment by danielcranford** [comment]
Yes please consider this. Interactive use of powershell is subpar without job control. Sure you can always spawn another terminal window, but that doesn't inherit your shell process state. You can also kill and restart the command in the background. But the ability to kick a command into the background and regain interactive use of the shell is incredibly powerful.

**Comment by superbonaci** [comment]
Maybe it's just not possible. The PowerShell can handle native apps but also NET and such, no idea if can be done.

**Comment by lost22git** [comment]
any progrees?

**Comment by SeeminglyScience** [comment]
The Engine WG discussed this. Due to the in-process nature of PowerShell commands and the reliance on blocking synchronous calls for the implementation of pipelines, we do not believe this feature is feasible to implement in a reliable and understandable manner.

**Comment by superbonaci** [comment]
Doesn't `Stop-Process` do just that?

**Comment by superbonaci** [comment]
Or maybe [PsSuspend](https://learn.microsoft.com/en-us/sysinternals/downloads/pssuspend) this is what I'm talking about.

**Comment by Lite5h4dow** [comment]
> > https://github.com/PowerShell/PowerShell/issues/3229#event-16500757975
> 
> This isn't completed.

Because they aren't going to implement it.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/3229#issuecomment-2688821169

@Lite5h4dow, that's made evident in the thread. What's not evident is why this has been closed as completed instead of unplanned.

**Comment by Lite5h4dow** [comment]
> > https://github.com/PowerShell/PowerShell/issues/3229#issuecomment-2688821169
> 
> @Lite5h4dow, that's made evident in the thread. What's not evident is why this has been closed as completed instead of unplanned.

Probably a missclick. Does it really matter?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/3229#issuecomment-2689012241

@Lite5h4dow, inasmuch as I commented. This seems like an X/Y problem – why are you asking?

**Comment by SeeminglyScience** [comment]
It's just the default for the bot that closes it, we don't use the field for anything so it's just meaningless text. I'll ask about having it changed though (and am manually changing this one)
